qiskit_experiments/warnings.py

```diff
@@ -0,0 +1,245 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A collection of typical warnings."""
+
+import functools
+import warnings
+
+from typing import Callable, Optional, Type, Dict
+
+
+def deprecated_logic(
+    version_removed: Optional[str] = None,
+    use_instead: Optional[str] = None,
```

Comment:
This is not very flexible since it asusmes there is a 1 to 1 replacement for the function. It would be better for having an option to just provide a custom msg as the first kwarg of this decorator
```suggestion
    msg: Optional[str] = None,
```